### ISSUE TITLE
Bug: Forge makes an unnecessary URL request when fetching `solc` build list.json information 

### ISSUE THREAD

**Issue by sambacha** [foundry-rs/foundry#7440]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (0026488 2024-03-19T00:20:40.934617000Z)

### Operating System

None

### Describe the bug

Forge makes a request to:

```
https://github.com/alloy-rs/solc-builds/raw/e4b80d33bc4d015b2fc3583e217fbf248b2014e1/macosx/aarch64/list.json
```

This hits a 302 re-direct to:

```
https://raw.githubusercontent.com/alloy-rs/solc-builds/e4b80d33bc4d015b2fc3583e217fbf248b2014e1/macosx/aarch64/list.json
```

Instead, why not just hit the `https://raw.githubusercontent.com` domain, instead of hitting `github.com` first? it will always be the case that this gets redirected.

**Comment by zerosnacks** [comment]
Should be resolved by https://github.com/alloy-rs/svm-rs/pull/132

**Comment by sambacha** [comment]
> Should be resolved by [alloy-rs/svm-rs#132](https://github.com/alloy-rs/svm-rs/pull/132)

My man, give this comrade a snack
